app/Domain/Tickets/Services/Tickets.php

```diff
@@ -2059,6 +2061,157 @@ public function prepareTicketDates(&$values)
+                'projectId' => $values['projectId'] ?? session("currentProject"),
+                'editorId' => $values['editorId'] ?? "",
+                'date' => dtHelper()->userNow()->formatDateTimeForDb(),
+                'dateToFinish' => $values['dateToFinish'] ?? "",
+                'timeToFinish' => $values['timeToFinish'] ?? "",
+                'status' => $values['status'] ?? "",
+                'planHours' => $values['planHours'] ?? "",
+                'tags' => $values['tags'] ?? "",
+                'sprint' => $values['sprint'] ?? "",
+                'storypoints' => $values['storypoints'] ?? "",
+                'hourRemaining' => $values['hourRemaining'] ?? "",
+                'priority' => $values['priority'] ?? "",
+                'acceptanceCriteria' => $values['acceptanceCriteria'] ?? "",
+                'editFrom' => $values['editFrom'] ?? "",
+                'editTo' => $values['editTo'] ?? "",
+                'dependingTicketId' => $values['dependingTicketId'] ?? "",
+                'milestoneid' => $values['milestoneid'] ?? "",
+            );
+
+            return $this->addTicket($values);
+        }
+
+        public function updateTodoUsingZapier($values): array
+        {
+            if (!isset($values['ticketId'])) {
+                error_log('we dont have a ticket ..');
+                return [
+                    "message" => "Missing ticketId",
+                    "status" => "error",
+                ];
+            }
+
+            $currentTicket = $this->getTicket($values['ticketId']);
+
+            if (!$currentTicket) {
+                return [
+                    "message" => "We do not have a ticket for id " . $values['ticketId'],
+                    "status" => "error",
+                ];
+            }
+
+            $values = array(
+                'id' => $values['ticketId'],
+                'headline' => $currentTicket->headline,
+                'editorId' => $values['editorId'] ?? "",
+                'status' => $values['status'] ?? "",
+                'milestoneid' => $values['milestoneid'] ?? "",
+            );
+
+            try {
+                $sucessfullyUpdated = $this->updateTicket($values);
+                if ($sucessfullyUpdated) {
+                    return [
+                        "message" => "Ticket updated successfully",
+                        "status" => "success",
+                    ];
+                } else {
+                    return [
+                        "message" => "Ticket could not be updated",
+                        "status" => "error",
+                    ];
+                }
+            } catch (\Exception $e) {
+                return [
+                    "message" => "Ticket could not be updated. Reason: " . $e->getMessage(),
+                    "status" => "error",
+                ];
+            }
+        }
+
+        public function createMilestoneUsingZapier($values): array|bool
```

Comment:
Is there anything special in here or could you just use the `addTicket` method from this file?

Reply:
yep I fully agree,

I am going to hard code the `type` to be `milestone` for the request body in zapier and utilize that `addTicket` function!